cpp/src/parquet/metadata.cc

```diff
@@ -132,6 +132,31 @@ std::shared_ptr<Statistics> MakeColumnStats(const format::ColumnMetaData& meta_d
   throw ParquetException("Can't decode page statistics for selected column type");
 }
 
+template <typename Metadata>
+std::shared_ptr<KeyValueMetadata> CopyKeyValueMetadata(const Metadata& source) {
+  std::shared_ptr<KeyValueMetadata> metadata = nullptr;
+  if (source.__isset.key_value_metadata) {
+    metadata = std::make_shared<KeyValueMetadata>();
+    for (const auto& it : source.key_value_metadata) {
+      metadata->Append(it.key, it.value);
+    }
+  }
+  return metadata;
+}
+
+template <typename Metadata>
+void ToThriftKeyValueMetadata(Metadata& metadata, const KeyValueMetadata& source) {
```

Comment:
We tend to avoid non-const references in the codebase, so I would suggest to use a pointer instead:
```suggestion
void ToThriftKeyValueMetadata(Metadata* metadata, const KeyValueMetadata& source) {
```